zerver/signals.py

```diff
@@ -73,15 +73,13 @@ def email_on_new_login(sender: Any, user: UserProfile, request: Any, **kwargs: A
         device_browser = get_device_browser(user_agent)
         device_os = get_device_os(user_agent)
         device_ip = request.META.get('REMOTE_ADDR') or "Uknown IP address"
-        device_info = {"device_browser": device_browser,
-                       "device_os": device_os,
-                       "device_ip": device_ip,
-                       "login_time": login_time
-                       }
 
         context = common_context(user)
-        context['device_info'] = device_info
         context['user_email'] = user.email
+        context['login_time'] = login_time
+        context['device_ip'] = device_ip
+        context['device_os'] = device_os
+        context['device_browser'] = device_browser
```

Comment:
I cleaned this up to get rid of the now-unnecsesary local variables for `device_browser` and friends and added a4ff91778931572e4d346d80c7f7f9dd73d7e9e0 to fix the "Uknown IP address" piece.